src/QMCWaveFunctions/RotatedSPOs.cpp

```diff
@@ -107,36 +111,45 @@ void RotatedSPOs::extractParamsFromAntiSymmetricMatrix(const RotationIndices& ro
 
 void RotatedSPOs::resetParametersExclusive(const opt_variables_type& active)
 {
-  std::vector<ValueType> delta_param(m_act_rot_inds_.size());
+  const size_t N = m_act_rot_inds_.size();
+  std::vector<ValueType> delta_param(N);
 
-  size_t psize = m_act_rot_inds_.size();
+  size_t psize = N;
 
   if (use_global_rot_)
   {
     psize = m_full_rot_inds_.size();
-    assert(psize >= m_act_rot_inds_.size());
+    assert(psize >= N);
   }
 
   std::vector<ValueType> old_param(psize);
   std::vector<ValueType> new_param(psize);
 
-  for (int i = 0; i < m_act_rot_inds_.size(); i++)
+  //cast ValueType to RealType for delta param
+  //allows us to work with both real and complex since
+  //active and myVars are stored as only reals
+  auto* delta_param_data_real = (RealType*)delta_param.data();
```

Comment:
There might be a better suffix than "_real"?  When I read it, my immediate thought  is that this vector is just the real part.   Maybe "_alias"?   Or put the cast and array reference on a single line to avoid the name altogether? (At the cost of complicating the left-hand side of the assignment)

Alternately, this pattern of copying data using a trick to use a single loop for both the real and complex cases is repeated several places. Would it be worthwhile to make  helper function to copy complex numbers to/from a real array?  (Maybe not for this particular loop, but the other loops where it appears)

Reply:
fixed